javalanglib/src/main/scala/java/lang/Character.scala

```diff
@@ -462,8 +462,54 @@ object Character {
   //def getDirectionality(c: scala.Char): scala.Byte
 
   /* Conversions */
-  def toUpperCase(c: scala.Char): scala.Char = c.toString.toUpperCase().charAt(0)
-  def toLowerCase(c: scala.Char): scala.Char = c.toString.toLowerCase().charAt(0)
+  def toUpperCase(ch: scala.Char): scala.Char = {
+    val upper = toUpperCase(ch.toInt)
+    if (upper > scala.Char.MaxValue) ch
+    else upper.toChar
+  }
+
+  def toLowerCase(ch: scala.Char): scala.Char = {
+    val lower = toLowerCase(ch.toInt)
+    if (lower > scala.Char.MaxValue) ch
+    else lower.toChar
+  }
+
+  def toUpperCase(codePoint: scala.Int): scala.Int = {
+    val upperChars = _String.fromCodePoint(codePoint).toUpperCase
+
+    if (upperChars.length > 2) {
+      codePoint
+    } else if (upperChars.length == 2) {
```

Comment:
Instead of this chain of if/else's, consider using a match:
```scala
upperChars.length match {
  case 1 =>
    upperChars.charAt(0).toInt
  case 2 =>
    ...
  case _ =>
    codePoint
}
```